File: 18_power_of_number/18_power_of_no.dart
import 'dart:io';

double powerOfNumber(int num, int pow) {
  double result = 1;

  int limit = pow.abs();

  for (int i = 0; i < limit; i++) {
    result = result * num;
  }

  if (pow < 0){
    return (1 / result);
  }

  return result;
}

void main() {
  try {
    while (true) {
      stdout.write("Enter a number : ");
      int num = int.parse(stdin.readLineSync()!);

      stdout.write("Enter a number : ");
      int pow = int.parse(stdin.readLineSync()!);

      if (num == 0 && pow == 0) {
        print("Enter positive integer value");
      }
      else {
        double result = powerOfNumber(num, pow);
        print(result);
      }

      stdout.write("Do you want to continue? (y/n) : ");
      String choice = stdin.readLineSync()!.toLowerCase();
      if (choice == 'n') {
        print("Program Finished");
        break;
      }
    }
  } on FormatException catch (e) {
    print("Invalid input : $e");
  } on Exception catch (e) {
    print("Exception caught : $e");
  }
}

File: 18_power_of_number/18_power_of_no.py
# python program to find the power of a number

# method 1 - using the operator
try:
    n=eval(input("Enter the number :"))
    m=int(input("Enter the power term :"))
    pow=n**m
    print(round(pow,4))
except:
    print('ENter a valid Integer value!')

# method 2 - using simple iteration
try:
    n=eval(input("Enter the number :"))
    m=int(input("Enter the power term :"))
    pow=1
    for i in range(m):
        pow=pow*n
    print(round(pow,4))
except:
    print("Enter a valid integer value!")

# method 3 - using recurssion
try:
    n=int(input("Enter a number :"))
    m=int(input("Enter a power term :"))
    def pow(a,b):
        if b==0:
            return 1
        return a*pow(a,b-1)
    print(pow(n,m))
except:
    print("Enter a valid integer value!")

# method 4 - using function based approach 

def power_of_number(num, pow):
    result = 1
    for i in range(1,pow+1):
        result = result * num
    return result

try:
    num = int(input("Enter number : "))
    pow = int(input("Enter power factor : "))
    if (num <= 0 or pow <= 0):
        print("Enter a positive value")
    else:
        result = power_of_number(num,  pow)
        print(result)

except ValueError as e:
    print("Invalid Input : {e}")

except Exception as e:
    print("Exception caught : {e}")


